### PR TITLE
Feature/hide app secret

### PR DESCRIPTION
**Pull request by jaelim-ms**

### CODE REVIEW

File: sdk/mobile-center-updates/src/main/java/com/microsoft/azure/mobile/updates/Updates.java [hunk 1]
```diff
@@ -602,7 +606,33 @@ private synchronized void getLatestReleaseDetails(@NonNull String updateToken) {
         Map<String, String> headers = new HashMap<>();
         headers.put(HEADER_API_TOKEN, updateToken);
         final Object releaseCallId = mCheckReleaseCallId = new Object();
-        mCheckReleaseApiCall = httpClient.callAsync(url, METHOD_GET, headers, null, new ServiceCallback() {
+        mCheckReleaseApiCall = httpClient.callAsync(url, METHOD_GET, headers, new HttpClient.CallTemplate() {
+
+            @Override
+            public String buildRequestBody() throws JSONException {
+
+                /* This method is getting called. */
```
Comment (guperrot) [inline sdk/mobile-center-updates/src/main/java/com/microsoft/azure/mobile/updates/Updates.java]: Actually this method is not called for a get.
I would either remove this comment or change it to something like "No request body as its a get call"

File: sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/http/HttpUtils.java [hunk 2]
```diff
@@ -65,4 +71,18 @@ public static boolean isRecoverableError(Throwable t) {
         }
         return false;
     }
+
+    public static String hideSecret(String secret) {
+
+        /* Cannot hide null or empty string. */
+        if (secret == null || secret.length() <= 0) {
```
Comment (guperrot) [inline sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/http/HttpUtils.java]: length cannot be negative. And you can use `.isEmpty()`

File: sdk/mobile-center-updates/src/main/java/com/microsoft/azure/mobile/updates/Updates.java (hunk 1 above)
Comment (jaelim-ms) [inline sdk/mobile-center-updates/src/main/java/com/microsoft/azure/mobile/updates/Updates.java]: Oh..  Where is my **NOT**????

File: sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/http/HttpUtils.java (hunk 2 above)
Comment (jaelim-ms) [inline sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/http/HttpUtils.java]: Using `isEmpty()` instead.

**Pull request merged**